plan/expression_rewriter.go

```diff
@@ -38,11 +47,59 @@ type expressionRewriter struct {
 	correlated bool
 }
 
-func (er *expressionRewriter) buildSubquery(subq *ast.SubqueryExpr) (Plan, expression.Schema) {
-	if len(er.b.outerSchemas) > 0 {
-		er.err = errors.New("Nested subqueries is not currently supported.")
-		return nil, nil
+func constructBinaryOpFunction(l *rowExpr, r *rowExpr, op opcode.Op) (*expression.ScalarFunction, error) {
+	if l.expr != nil && r.expr != nil {
+		return expression.NewFunction(opcode.Ops[op], []expression.Expression{l.expr, r.expr}, types.NewFieldType(mysql.TypeTiny))
+	} else if l.expr != nil {
+		return nil, errors.New("Operand should contain 1 column(s)")
+	} else if r.expr != nil || len(l.rows) != len(r.rows) {
+		return nil, errors.Errorf("Operand should contain %d column(s)", len(l.rows))
+	}
+	result, err := constructBinaryOpFunction(l.rows[0], r.rows[0], op)
```

Comment:
We can append result to a slice, and use `composeConditions` function to compose them.